### PR TITLE
mgr/dashboard: Separate file for icons for refering and reusing

### PR DESCRIPTION
**Pull request by a2batic**
Fixes: http://tracker.ceph.com/issues/39032
Signed-off-by: Kanika Murarka <kmurarka@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/shared/enum/icons.enum.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  destroy = 'fa fa-times', // Destroy, Remove
+  exchange = 'fa fa-exchange', // Edit Peer
+  copy = 'fa fa-copy', // Copy
+  flatten = 'fa-chain-broken', // Flattern, Link broken
+  trash = 'fa fa-trash-o', // Delete, Move to trash
+  lock = 'fa fa-lock', // Protect
+  unlock = 'fa fa-unlock', // Unprotect
+  clone = 'fa fa-clone', // clone
+  undo = 'fa fa-undo', // Rollback
+  search = 'glyphicon glyphicon-search', // Search
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/enum/icons.enum.ts]:
```suggestion
  search = 'fa fa-search', // Search
```

Comment (a2batic) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/enum/icons.enum.ts]: ack, Thanks

**Comment by a2batic** [comment]
jenkins retest please

**Comment by a2batic** [comment]
jenkins test make check

**Comment by a2batic** [comment]
jenkins retest this please

File: src/pybind/mgr/dashboard/frontend/src/app/core/forbidden/forbidden.component.ts [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 
 @Component({
   selector: 'cd-forbidden',
   templateUrl: './forbidden.component.html',
   styleUrls: ['./forbidden.component.scss']
 })
 export class ForbiddenComponent {
-  constructor() {}
+  icons: any;
+  constructor() {
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/src/app/core/forbidden/forbidden.component.ts]: Alternative approach:

`export class ForbiddenComponent {`
`  @Input() icons = Icons;`
`}`

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html [hunk 3]
```diff
@@ -32,7 +32,7 @@
                         id="ecp-info-button"
                         type="button"
                         (click)="targetSettingsModal()">
-                  <i class="fa fa-cogs fa-fw"
+                  <i class="{{ icons.deepCheck + ' ' + icons.width }}"
```
[the remark this answers is not in the document]
Comment (a2batic) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.html]: ack

File: src/pybind/mgr/dashboard/frontend/src/app/core/forbidden/forbidden.component.ts (hunk 2 above)
Comment (a2batic) [reply src/pybind/mgr/dashboard/frontend/src/app/core/forbidden/forbidden.component.ts]: ack

**Comment by a2batic** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.html [hunk 4]
```diff
@@ -5,7 +5,7 @@
     <h3 class="page-header">
       <span
         (click)="toggleSectionVisibility(section.class)"
-        class="collapsible">{{ section.heading }} <i [ngClass]="{'fa-plus-circle': !sectionVisibility[section.class], 'fa-minus-circle': sectionVisibility[section.class]}" class="fa" aria-hidden="true"></i></span>
+        class="collapsible">{{ section.heading }} <i [ngClass]="{ 'fa-plus-circle' : !sectionVisibility[section.class], 'fa-minus-circle': sectionVisibility[section.class]}" class="fa" aria-hidden="true"></i></span>
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.html]:
```suggestion
        class="collapsible">{{ section.heading }} <i [ngClass]="{ icons.addCircle : !sectionVisibility[section.class], 'fa-minus-circle': sectionVisibility[section.class]}" class="fa" aria-hidden="true"></i></span>
```
Not sure if `fa-minus-circle` should be replaced with another icon or added to the list of icons...

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.ts [hunk 5]
```diff
@@ -25,6 +26,9 @@ export class RbdConfigurationFormComponent implements OnInit {
   }>;
   @Output()
   changes = new EventEmitter<any>();
+
+  @Input() icons = Icons;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.ts]:
```suggestion
  icons = Icons;
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-trash-list/rbd-trash-list.component.ts [hunk 6]
```diff
@@ -37,6 +38,8 @@ export class RbdTrashListComponent implements OnInit {
   @ViewChild('deleteTpl')
   deleteTpl: TemplateRef<any>;
 
+  @Input() icons = Icons;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-trash-list/rbd-trash-list.component.ts]:
```suggestion
  icons = Icons;
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-trash-list/rbd-trash-list.component.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { Component, OnInit, TemplateRef, ViewChild } from '@angular/core';
+import { Component, Input, OnInit, TemplateRef, ViewChild } from '@angular/core';
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-trash-list/rbd-trash-list.component.ts]:
```suggestion
import { Component, OnInit, TemplateRef, ViewChild } from '@angular/core';
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts [hunk 8]
```diff
@@ -1,9 +1,10 @@
-import { Component, OnInit, TemplateRef, ViewChild } from '@angular/core';
+import { Component, Input, OnInit, TemplateRef, ViewChild } from '@angular/core';
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts]:
```suggestion
import { Component, OnInit, TemplateRef, ViewChild } from '@angular/core';
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts [hunk 9]
```diff
@@ -20,6 +21,7 @@ export class ConfigurationComponent implements OnInit {
   permission: Permission;
   tableActions: CdTableAction[];
   data = [];
+  @Input() icons = Icons;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/configuration/configuration.component.ts]:
```suggestion
  icons = Icons;
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/logs/logs.component.ts [hunk 10]
```diff
@@ -11,6 +12,7 @@ export class LogsComponent implements OnInit, OnDestroy {
   contentData: any;
   clog: Array<any>;
   audit_log: Array<any>;
+  @Input() icons = Icons;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/logs/logs.component.ts]:
```suggestion
  icons = Icons;
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts [hunk 11]
```diff
@@ -45,7 +46,7 @@ export class OsdListComponent implements OnInit {
   tableActions: CdTableAction[];
   bsModalRef: BsModalRef;
   columns: CdTableColumn[];
-
+  @Input() icons = Icons;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts]:
```suggestion
  icons = Icons;
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/prometheus/prometheus-list/prometheus-list.component.ts [hunk 12]
```diff
@@ -16,6 +17,7 @@ export class PrometheusListComponent implements OnInit {
   externalLinkTpl: TemplateRef<any>;
   columns: CdTableColumn[];
   selection = new CdTableSelection();
+  @Input() icons = Icons;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/prometheus/prometheus-list/prometheus-list.component.ts]:
```suggestion
  icons = Icons;
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard/health/health.component.ts [hunk 13]
```diff
@@ -25,6 +26,7 @@ export class HealthComponent implements OnInit, OnDestroy {
   interval = new Subscription();
   permissions: Permissions;
   enabledFeature$: FeatureTogglesMap$;
+  @Input() icons = Icons;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard/health/health.component.ts]:
```suggestion
  icons = Icons;
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-form-client/nfs-form-client.component.ts [hunk 14]
```diff
@@ -18,6 +19,7 @@ export class NfsFormClientComponent {
 
   nfsSquash: any[] = this.nfsService.nfsSquash;
   nfsAccessType: any[] = this.nfsService.nfsAccessType;
+  @Input() icons = Icons;
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-form-client/nfs-form-client.component.ts]:
```suggestion
  icons = Icons;
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.html (hunk 4 above)
Comment (a2batic) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.html]: I might have missed it, thanks

**Comment by a2batic** [comment]
jenkins retest please

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.html (hunk 4 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.html]:
```suggestion
        class="collapsible">{{ section.heading }} <i [ngClass]="!sectionVisibility[section.class] ? icons.addCircle : icons.minusCircle" class="fa" aria-hidden="true"></i></span>
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-details/iscsi-target-details.component.spec.ts [hunk 15]
```diff
@@ -97,15 +97,18 @@ describe('IscsiTargetDetailsComponent', () => {
         {
           children: [{ id: 'disk_rbd_disk_1', value: 'rbd/disk_1' }],
           settings: {
-            cssClasses: { expanded: 'fa fa-fw fa-hdd-o fa-lg', leaf: 'fa fa-fw fa-hdd-o' },
+            cssClasses: { expanded: 'fa fa-fw fa fa-lg fa fa-hdd-o', leaf: 'fa fa-fw fa fa-hdd-o' },
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-details/iscsi-target-details.component.spec.ts]: What about using this?
```suggestion
            cssClasses: { expanded: _.join([Icons.width, Icons.large, Icons.disk], ' '), leaf: _.join([Icons.width, Icons.disk], ' ') },
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-details/iscsi-target-details.component.ts [hunk 16]
```diff
@@ -75,26 +76,25 @@ export class IscsiTargetDetailsComponent implements OnChanges, OnInit {
 
   private generateTree() {
     this.metadata = { root: this.selectedItem.target_controls };
-
     const cssClasses = {
       target: {
-        expanded: 'fa fa-fw fa-bullseye fa-lg'
+        expanded: Icons.width + ' ' + Icons.large + ' ' + Icons.bullseye
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-details/iscsi-target-details.component.ts]: What about using this?
```suggestion
        expanded: _.join([Icons.width, Icons.large, Icons.bullseye], ' ')
```

**Comment by tspmelo** [comment]
jenkins test dashboard

**Pull request merged**